tests/generative/variable_strategies.py

```diff
@@ -362,6 +364,33 @@ def binary_operation_with_types(
 
         return operator_func(lhs, rhs)
 
+    @st.composite
+    def nary_operation_with_types(draw, frame, operator_func, series_type):
+        # A strategy for operations that take _n_ arguments which are expected to be
+        # the same type
+
+        # A nary operation has n args, which are
+        # 1) A number of series drawn from the provided frame or another frame,
+        # or series that are Values
+
+        # first decide on a non-zero value for n within a reasonable range
+        n = draw(st.integers(1, 20))
+
+        # draw _n_ args of same type
+        series_options = [value, series]
+        args = []
+        for _ in range(n):
+            arg_frame = draw(st.one_of(any_frame(), st.just(frame)))
+            arg_series = draw(st.sampled_from(series_options))
+            args.append(draw(arg_series(series_type, arg_frame)))
```

Comment:
I think this could be generated using a hypothesis `lists` strategy; the contents of the loop would be a strategy for generating a nary arg, and the `lists` strategy has a min/max size arg that would deal with picking the number of input args.

Something like:
```
@st.composite
def narg_strategy():
    series_options = [value, series]
    arg_frame = draw(st.one_of(any_frame(), st.just(frame))
    return draw(st.sampled_from(series_options))

args = st.lists(narg_strategy(), min_size=1, max_size=20)
```

(not sure if that'd actually work, I always seem to need to experiment with where `composite` and `draw` go)

Reply:
that looks cleaner, I'll have a poke at it